### DISCUSSION TITLE
What is the correct method to scale a shape?

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics.js#171] [Q&A] [answered]
I am working on resizing bodies/shapes. 
I know shapes can't change their size dynamically but need to be recreated.

In examples I saw scaled shapes were wrapped in decorator shapes: [ScaledShape](https://jrouwe.github.io/JoltPhysics/class_scaled_shape.html)

However, I see the core `Jolt.Shape` also has a `.ScaleShape()` method. [(here)](https://jrouwe.github.io/JoltPhysics/class_shape.html#ae4a0954ec23eabd48f0961b16eff750a)

I do see the disclaimer on the instance method that not all shapes can be scaled, so I'm wondering if that makes the `ScaledShape` decorator the more stable method.

My current setup saves the `ShapeSettings` and when called generates a new `Shape` AND `ScaledShape`

The method pattern isn't really simpler code wise but is a different pattern.

Is there any benefit one way or the other?

**Comment by jrouwe** [comment, marked as answer]
I have created some documentation and a demo here: https://jrouwe.github.io/JoltPhysics/#scaling-shapes (note in C++, concepts should work in JS too)

**Reply by DennisSmolek** [reply]
Ah gotcha, so it's the other way around.


With the `SetScale` method being safer, what is the usecase for the `ScaledShape` decorator?

**Reply by jrouwe** [reply]
ScaleShape() uses ScaledShape internally
